src/scripts/main.js

```diff
@@ -1,3 +1,289 @@
+
+    if (field === 'Age' || field === 'Salary') {
+      input.setAttribute('type', 'number');
+    } else {
+      input.setAttribute('type', 'text');
+    }
+
+    if (field === 'Age') {
+      input.setAttribute('min', 18);
+      input.setAttribute('max', 90);
+    }
+
+    label.append(input);
+    fragment.append(label);
+  }
+});
+
+const button = document.createElement('button');
+
+button.textContent = 'Save to table';
+
+fragment.append(button);
+form.append(fragment);
+document.body.firstElementChild.after(form);
+
+const formBtn = form.querySelector('button');
+
+function addSeparator(num) {
+  return num.toLocaleString('en-US');
+}
+
+function CustomNotification(title, description, type) {
+  this.title = title;
+  this.description = description;
+  this.type = type;
+}
+
+const pushNotification = ({ title, description, type }) => {
+  const notification = document.createElement('div');
+
+  notification.classList.add('notification', `${type}`);
+  notification.style.top = 15 + 'px';
+  notification.style.right = 15 + 'px';
+
+  const fragmentNot = document.createDocumentFragment();
+  const titleNot = document.createElement('h2');
+
+  titleNot.classList.add('title');
+  titleNot.textContent = title;
+
+  const descriptionNot = document.createElement('p');
+
+  descriptionNot.textContent = description;
+
+  fragmentNot.append(titleNot);
+  fragmentNot.append(descriptionNot);
+  notification.append(fragmentNot);
+
+  document.body.append(notification);
+
+  setTimeout(() => {
+    notification.hidden = true;
+  }, 2000);
+};
+
+formBtn.addEventListener('click', (e) => {
+  e.preventDefault();
+
+  const formValues = {
+    employeeName: form.elements.name.value,
+    position: form.elements.position.value,
+    office: form.elements.office.value,
+    age: form.elements.age.value,
+    salary: form.elements.salary.value.trim(),
+  };
+
+  const formSalaryValue =
+    !formValues.salary || isNaN(formValues.salary)
+      ? ''
+      : '$' + addSeparator(+formValues.salary);
+
+  const fields = validateFormFields(formValues);
+
+  pushNotification(fields[0]);
+
+  if (fields[0].type === 'success') {
+    const row = addRow([
+      formValues.employeeName,
+      formValues.position,
+      formValues.office,
+      formValues.age,
+      formSalaryValue,
+    ]);
+
+    tbody.append(row);
+  }
+});
+
+function validateFormFields({ employeeName, position, office, age, salary }) {
+  const notifications = [];
+
+  if (employeeName.length < 4) {
+    notifications.push(
+      new CustomNotification(
+        'Error',
+        'Name must be at least 4 characters long',
+        'error',
+      ),
+    );
+  }
+
+  if (!position.trim()) {
+    notifications.push(
+      new CustomNotification('Error', 'Position cannot be empty', 'error'),
+    );
+  }
+
+  if (!office.trim() || !offices.includes(office)) {
+    notifications.push(
+      new CustomNotification('Error', 'Office cannot be empty', 'error'),
+    );
```

Comment:
The validation for the 'Office' field only checks if it is non-empty and if it is included in the 'offices' array. However, the error message only mentions that the office cannot be empty. Consider updating the error message to reflect that the office must be one of the predefined options.

Reply:
"If the AI previously approved the changes, but some inaccuracies were still noted, I made the necessary corrections and submitted a new pull request. However, the review didn't take place; instead, the new commit was immediately marked as 'rejected' without any explanation. Should I expect a review, or does the AI-mentor stop responding to requests after the initial approval, and further assistance is no longer available?"